kahuna/public/js/search/query.js

```diff
@@ -101,7 +104,7 @@ query.controller('SearchQueryCtrl',
     });
 
     function resetQueryAndFocus() {
-        ctrl.filter.query = '';
+        ctrl.filter.query = undefined;
```

Comment:
I don't think so as `''` matches the value set when there is an empty query param. I think deleting it would stop it matching and trigger another state.go.